### PR TITLE
[PLATFORM-282] Stream editor: Historical data

### PR DESCRIPTION
**Pull request by tumppi**
Add historical data section to stream editor. This allows uploading a CSV file with stream data and also deleting events from the stream.

Uses new CSV upload API from https://github.com/streamr-dev/engine-and-editor/pull/505.

### CODE REVIEW

**Comment by tumppi** [comment]
@juhah: You can use e.g. this file:

_Timestamp,Data
01/01/2019 11:00:00.000,1
02/01/2019 12:00:00.000,2
03/01/2019 13:00:00.000,3
04/01/2019 14:00:00.000,4
05/01/2019 15:00:00.000,5_

**Comment by fonty1** [comment]
@tumppi should this be working on the Create Stream page? I can't seem to activate the modal on a dragndrop there yet.

File: app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx [hunk 1]
```diff
@@ -2,99 +2,258 @@
...
+
+        if (streamId && csvFile) {
+            uploadCsvFile(streamId, csvFile)
+                .then(() => {
+                    // Old API would accept the file as it is but now we
+                    // have to confirm the fields
+                    this.openConfigurationModal()
+                })
+                .catch(() => {
+                    this.openConfigurationModal()
```
Comment (juhah) [inline app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx]: This catches the error thrown in line 410 of `app/src/userpages/modules/userPageStreams/actions.js`:

```
if (response.data.schema.timestampColumnIndex == null) {
    dispatch(uploadCsvFileUnknownSchema(id, response.data.fileId, response.data.schema))
    throw new Error('Could not parse timestamp column!')
}
```
Is that desired behaviour? Maybe we should add a specific error type and catch it here and throw anything beside that?

File: app/src/userpages/modules/userPageStreams/actions.js [hunk 2]
```diff
@@ -396,32 +404,32 @@ export const uploadCsvFile = (id: StreamId, file: File) => (dispatch: Function)
         },
         withCredentials: true,
     })
-        .then(() => {
-            dispatch(uploadCsvFileSuccess())
+        .then((response) => {
+            if (response.data.schema.timestampColumnIndex == null) {
+                dispatch(uploadCsvFileUnknownSchema(id, response.data.fileId, response.data.schema))
+                throw new Error('Could not parse timestamp column!')
```
Comment (juhah) [inline app/src/userpages/modules/userPageStreams/actions.js]: This marks the upload as failed as it goes to the `catch` block but the upload still opens the confirm modal.

File: app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx [hunk 3]
```diff
@@ -2,99 +2,258 @@
...
+                                preserveErrorSpace
+                            />
+                        </Col>
+                        <Col {...rightColumn}>
+                            <Button
+                                className={styles.deleteButton}
+                                color="userpages"
+                                onClick={() => this.deleteDataUpTo(streamId, deleteDate)}
+                            >
+                                <Translate value="userpages.streams.edit.history.delete" />
```
[the remark this answers is not in the document]
Comment (juhah) [reply app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx]: Also, it should probably be disabled before a date is selected.

File: app/src/userpages/components/StreamPage/ConfirmCsvImportDialog/index.jsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                        cancel: {
+                            title: I18n.t('modal.common.cancel'),
+                            onClick: onClose,
+                            color: 'link',
+                        },
+                        confirm: {
+                            title: I18n.t('modal.common.confirm'),
+                            color: 'primary',
+                            outline: true,
+                            onClick: this.onConfirm,
```
Comment (juhah) [inline app/src/userpages/components/StreamPage/ConfirmCsvImportDialog/index.jsx]: You could add a wait state to the button to see if the confirm was successful.

File: app/src/userpages/components/StreamPage/constants.js [hunk 5]
```diff
@@ -11,4 +11,4 @@ export const rightColumn = {
     },
 }
 
-export const fieldTypes = ['number', 'boolean', 'map', 'list', 'string']
+export const fieldTypes = ['number', 'boolean', 'map', 'list', 'string', 'timestamp']
```
Comment (fonty1) [inline app/src/userpages/components/StreamPage/constants.js]: Should this be `date` instead of `timestamp`?

**Comment by tumppi** [comment]
@fonty1: Good point! Backend needs the stream to be created to upload data :( Maybe I'll just add a message for the user to save the stream first.

File: app/src/userpages/components/StreamPage/constants.js (hunk 5 above)
Comment (tumppi) [reply app/src/userpages/components/StreamPage/constants.js]: Backend uses `timestamp` as the type identifier.

File: app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx (hunk 1 above)
Comment (tumppi) [reply app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx]: Yeah, It's desired behaviour but I agree that specific error type should be added 👍

**Pull request merged**